src/Common/Method/GetterReturnTypeRule.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace ArchitectureSniffer\Common\Method;
+
+use PDepend\Source\AST\ASTCallable;
+use PHPMD\AbstractNode;
+use PHPMD\AbstractRule;
+use PHPMD\Node\ClassNode;
+use PHPMD\Node\MethodNode;
+use PHPMD\Rule\ClassAware;
+
+class GetterReturnTypeRule extends AbstractRule implements ClassAware
+{
+    /**
+     * @var string
+     */
+    protected const PATTER_RETURN = '/@return\s+([^\s]+)/';
+
+    /**
+     * @var string
+     */
+    protected const PATTERN_INHERIT_DOC = '/@inheritDoc/i';
+
+    /**
+     * @var string
+     */
+    protected const PREFIX_METHOD_NAME = 'get';
+
+    /**
+     * @var string
+     */
+    protected const RULE_RETURN_TYPE_SPECIFIED = 'Getter `%s` must return something. Please add return type.';
+
+    /**
+     * @var string
+     */
+    protected const RULE_RETURN_NOT_BOOL = 'Getter `%s` must not return bool.';
+
+    /**
+     * @param \PHPMD\AbstractNode $node
+     *
+     * @return void
+     */
+    public function apply(AbstractNode $node)
+    {
+        if (!$node instanceof ClassNode) {
+            return;
+        }
+
+        foreach ($node->getMethods() as $methodNode) {
+            $this->verifyMethod($methodNode);
+        }
+    }
+
+    /**
+     * @param \PHPMD\Node\MethodNode $methodNode
+     *
+     * @return void
+     */
+    protected function verifyMethod(MethodNode $methodNode): void
+    {
+        if (strpos($methodNode->getName(), static::PREFIX_METHOD_NAME) !== 0) {
+            return;
+        }
+
+        $returnType = $this->getReturnType($methodNode);
+
+        if (
+            $returnType === null && !$this->isInheritDocBlockExist($methodNode)
+            || $returnType === 'void'
+        ) {
+            $this->addMethodMustReturnViolation($methodNode);
+
+            return;
+        }
+
+        if ($returnType === 'bool') {
+            $this->addMethodReturnBoolViolation($methodNode);
+        }
+    }
+
+    /**
+     * @param \PHPMD\Node\MethodNode $methodNode
+     *
+     * @return bool
+     */
+    protected function isInheritDocBlockExist(MethodNode $methodNode): bool
```

Comment:
Waah
Please also add sniff for this naming convention violates
Never ever use is Exist

Reply:
Added. Should I update `architecture-baseline.json` for all modules?